### ISSUE TITLE
Emit unused-variable for unused exceptions in exception handlers

### ISSUE THREAD

**Issue by sushobhit27** [PyCQA/pylint#2223]
### Steps to reproduce
1. cat sample.py
```python
try:
    a = 1/0
except ArithmeticError as e:
    pass
```
2. pylint sample.py
### Current behavior
--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)

### Expected behavior
pylint should give a warning that only pass statement is in the except block, as the exception is simply consumed without doing meaningful operation on it.
This issue occurs occasionally, as I personally write skeleton try/except block with pass and later on add either logging or re raising exception code in except block after completing the main code in try block. However sometimes I just miss some except blocks and this message should highlight this issue.

Note: Although this is sometimes expected behavior, but if not, then can lead to subtle bugs.
```python
for key in some_seq:
    try:
        print(d[key])
    except KeyError as ke:
        pass
```
### pylint --version output

pylint 2.0.0.dev1
astroid 2.0.0.dev3
Python 3.6.4 (default, Jan  7 2018, 15:53:53)
[GCC 6.4.0]

Another similar issue but somewhat minor than above one is:

### Steps to reproduce
1. cat sample.py
```python
try:
    a = 1/0
except ArithmeticError as e:
    print('ArithmeticError occurred')
```
2. pylint sample.py
### Current behavior
--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)

### Expected behavior
pylint should give a warning that variable 'e' is not used in the except block.

**Comment by PCManticore** [comment]
This is intended. We're only emitting in some conditions for `except:` and `except Exception:` but for general exception handling we're not doing anything special as we cannot know the intention of the user. Just imagine if we'd introduce a warning for `except ...: pass`, we'll suddenly introduce hundreds of violations in projects for code that might not be buggy at all.

What should we fix though is the second example, we should emit that `e` is unused.

**Comment by sushobhit27** [comment]
> Just imagine if we'd introduce a warning for except ...: pass, we'll suddenly introduce hundreds of violations in projects for code that might not be buggy at all.
hmm that's true, that's why I also mentioned the valid case.

**Comment by sushobhit27** [comment]
@PCManticore is there any function like is_name_used_in_scope in checkers/utils.py?

**Comment by jacobtylerwalls** [comment]
Duplicate of #626, which was fixed. Suggest closing.

**Comment by Pierre-Sassoulas** [comment]
@jacobtylerwalls it looks like it's not exactly the same thing, we still do not emit warning for the snippet provided in the issue:
````
$ cat a.py 
try:
    a = 1/0
except ArithmeticError as e:
    print('ArithmeticError occurred')
$ pylint a.py --enable=all --enable-all-extensions
************* Module a
a.py:1:0: C0114: Missing module docstring (missing-module-docstring)

------------------------------------------------------------------
Your code has been rated at 7.50/10 (previous run: 7.50/10, +0.00)
```

**Comment by jacobtylerwalls** [comment]
True, thank you. Closely related to #4391.

**Comment by jacobtylerwalls** [comment]
These examples were only presented at the module-level. These work with `--allow-global-unused-variables=False` in your config.
```python
try:
    a = 1/0
except ArithmeticError as e:
    print('ArithmeticError occurred')
```
```
% pylint a.py --allow-global-unused-variables=False
************* Module a
a.py:2:4: W0612: Unused variable 'a' (unused-variable)
a.py:3:0: W0612: Unused variable 'e' (unused-variable)
```

For function examples, see discussion at #4391 about whether we need more handling.
